test/dag-link-test.spec.js

```diff
@@ -58,9 +59,4 @@ describe('DAGLink', () => {
     const link = new DAGLink('hello', 3, cid)
     expect(link.Hash.toBaseEncodedString()).to.equal(cid)
   })
-
-  it('has an immutable CID', () => {
```

Comment:
Is this a breaking change because it makes sense or because of breaking things while we can?

Reply:
Using `Object.defineProperties` makes it had to infer what's a class property and what's a dynamically assigned property.

Immutability of programatic structures at this level doesn't really give us much so it seemed simpler to remove it..